src/pubsub.js

```diff
@@ -245,73 +209,45 @@ class BasicPubSub extends Pubsub {
   }
 
   _handleRpcControl (peer, rpc) {
-    throw errcode('_handleRpcControl must be implemented by the subclass', 'ERR_NOT_IMPLEMENTED')
+    throw errcode(new Error('_handleRpcControl must be implemented by the subclass'), 'ERR_NOT_IMPLEMENTED')
   }
 
   /**
    * Returns a buffer of a RPC message that contains a control message
-   *
    * @param {Array<rpc.RPC.Message>} msgs
    * @param {Array<rpc.RPC.ControlIHave>} ihave
    * @param {Array<rpc.RPC.ControlIWant>} iwant
    * @param {Array<rpc.RPC.ControlGraft>} graft
    * @param {Array<rpc.RPC.Prune>} prune
-   *
    * @returns {rpc.RPC}
-   *
    */
-  _rpcWithControl (msgs, ihave, iwant, graft, prune) {
+  _rpcWithControl (msgs = [], ihave = [], iwant = [], graft = [], prune = []) {
```

Comment:
I think we need to change the calling sites from `rpcWithControl(foo, null, bar, null, null)` to  `rpcWithControl(foo, undefined, bar, undefined, undefined)` to get the same behavior as before.

But the fact that most of the tests pass make me think we may not even need this?